server/src/main/java/org/opensearch/cluster/coordination/JoinTaskExecutor.java

```diff
@@ -170,12 +181,22 @@ public ClusterTasksResult<Task> execute(ClusterState currentState, List<Task> jo
         // processing any joins
         Map<String, String> joiniedNodeNameIds = new HashMap<>();
         for (final Task joinTask : joiningNodes) {
+            final DiscoveryNode node = joinTask.node();
             if (joinTask.isBecomeClusterManagerTask() || joinTask.isFinishElectionTask()) {
                 // noop
```

Comment:
Why is it a noop? Can a finish election task/become cluster manager task not have remote store node joining nodes?